### PR TITLE
[DeadCode] Add RemoveDeadLoopRector

### LINKED ISSUE #5513: [DeadCode] Add dead for each/while rule
E.g. empty foreach with no body

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #5513

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/dead-code/src/Rector/For_/RemoveDeadLoopRector.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run($values)
+    {
+        for ($i=1; $i<count($values); ++$i) {
+        }
+    }
+}
```
Comment (TomasVotruba) [inline rules/dead-code/src/Rector/For_/RemoveDeadLoopRector.php:46]:
```suggestion
class SomeClass
{
    public function run($values)
    {
    }
}
```

**Comment by samsonasik** [comment]
@TomasVotruba code example updated

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thanks :+1:

**Pull request merged**